src/utils/utils.rs

```diff
@@ -611,7 +611,9 @@ where
 }
 
 pub(crate) fn delete_dir_contents(dir_path: &Path) {
```

Comment:
this looks pretty good. I think perhaps renaming this function as well (or having a non-following-version and a following-version for callers where we don't want to escape out of our trees) would be a good idea, but I haven't checked in the code where delete_dir_contents is invoked today.

Reply:
The only caller is `rustup_mode::update()`, which calls this directly with `cfg.download_dir` and via `temp::Cfg`, which calls this with `temp::Cfg::root_directory`.

Reply:
@rbtcollins I've changed the function name accordingly.